libazureinit/src/provision/ssh.rs

```diff
@@ -333,7 +345,9 @@ mod tests {
             },
         ];
 
-        provision_ssh(&user, &keys, Some(".ssh/xauthorized_keys".to_string()))
+        let authorized_keys_path = user.dir.join(".ssh/xauthorized_keys");
+
+        provision_ssh(&user, &keys, authorized_keys_path.clone(), false)
```

Comment:
Why is `authorized_keys_path` being cloned? Without `.clone()` it seems to work fine.